middleware/middleware-rabbitmq.go

```diff
@@ -0,0 +1,199 @@
+package middleware
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/ptourne/sistemas-distribuidos-1/common"
+	"github.com/ptourne/sistemas-distribuidos-1/common/logger"
+	amqp "github.com/rabbitmq/amqp091-go"
+)
+
+
+type MiddlewareRabbitmq struct {
+	Conn *amqp.Connection
+	Ch    *amqp.Channel
+	readChannels map[string]*<-chan amqp.Delivery
+}
+
+var log = logger.NewConsoleLogger("middleee", logger.Debug)
+
+
+func NewMiddlewareRabbitmq() (*MiddlewareRabbitmq, error) {
+	conn, err := amqp.Dial("amqp://guest:guest@rabbitmq:5672/")
+	for range 5 {
+		if err == nil {
+			break
+		}
+		log.Errorf("Failed to connect to RabbitMQ: %v", err)
+		time.Sleep(5 * time.Second)
+		log.Infof("Retrying connection...")
+		conn, err = amqp.Dial("amqp://guest:guest@rabbitmq:5672/")
+	}
+	if err != nil {
+		return nil, err
+	}
+	ch, err2 := conn.Channel()
+	if err2 != nil {
+		return nil, fmt.Errorf("failed to open a channel: %v", err2)
+	}
+	middleware:= MiddlewareRabbitmq{
+		Conn: conn,
+		Ch:    ch,
+		readChannels: map[string]*<-chan amqp.Delivery{},
+	};
+	return &middleware, nil
+}
+
+func (m *MiddlewareRabbitmq) Close() error {
+	log.Infof("CLOSING")
+	if m.Ch != nil {
+		m.Ch.Close()
+	}
+	if m.Conn != nil {
+		m.Conn.Close()
+	}
```

Comment:
Podríamos setearlos a nil para evitar un Double Close accidental
```suggestion
	if m.Ch != nil {
		m.Ch.Close()
	        m.Ch = nil
	}
	if m.Conn != nil {
		m.Conn.Close()
		m.Conn = nil
	}
```